### ISSUE TITLE
AttributeError in extension "private_import" (C2701)

### ISSUE THREAD

**Issue by hofrob** [PyCQA/pylint#6624]
### Bug description

Basic example repo: https://github.com/hofrob/pylint-no-attribute-expr

```shell
pip install -r requirements.txt
pylint main.py
```

The code I want to check:

```py
from typing import Optional


class Bar:
    @property
    def get_foo(self):
        foo: Foo = Foo().save()
        return 1


class Foo:
    def save(self):
        return self
```

The import is necessary. It doesn't matter what's imported. But without it, the error is not triggered.

### Configuration

```ini
[MASTER]
load-plugins=pylint.extensions.private_import

[MESSAGES CONTROL]
disable=all
enable=C2701
```

### Command used

```shell
pylint main.py
```

### Pylint output

```shell
$ pylint main.py
Exception on node <ImportFrom l.1 at 0x7fe510f003d0> in file '../pylint-no-attribute-expr/main.py'
Traceback (most recent call last):
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 73, in walk
    callback(astroid)
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 67, in visit_importfrom
    private_names = self._get_type_annotation_names(node, names)
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 117, in _get_type_annotation_names
    self._populate_type_annotations(node.root(), self.all_used_type_annotations)
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 159, in _populate_type_annotations
    self._populate_type_annotations(
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 159, in _populate_type_annotations
    self._populate_type_annotations(
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 166, in _populate_type_annotations
    ] = self._assignments_call_private_name(name_assignments, private_name)
  File "../pylint-no-attribute-expr/venv/lib/python3.10/site-packages/pylint/extensions/private_import.py", line 232, in _assignments_call_private_name
    current_attribute = current_attribute.expr
AttributeError: 'Name' object has no attribute 'expr'

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

No AttributeError.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
```

### OS / Environment

Ubuntu 22.04

### Additional dependencies

astroid==2.11.5
dill==0.3.4
isort==5.10.1
lazy-object-proxy==1.7.1
mccabe==0.7.0
platformdirs==2.5.2
pylint==2.13.9
tomli==2.0.1
wrapt==1.14.1

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I cannot reproduce with python 3.8, but it might be specific to python 3.10.

**Comment by mbyrnepr2** [comment]
I can reproduce with:
```python
(venv310) mytest pylint --version
pylint 2.13.9
astroid 2.11.5
Python 3.10.4 (v3.10.4:9d38120e33, Mar 23 2022, 17:29:05) [Clang 13.0.0 (clang-1300.0.29.30)]
(venv310) mytest python -V
Python 3.10.4
```

**Comment by jacobtylerwalls** [comment]
I can reproduce with:
```shell
pylint 2.15.0-dev0
astroid 2.12.0-dev0
Python 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 09:09:08) 
[Clang 12.0.5 (clang-1205.0.22.9)]
```
